### PR TITLE
app-editors/nvi: add snapshot

### PR DESCRIPTION
**Pull request by doughdemon**

### CODE REVIEW

File: app-editors/nvi/nvi-1.81.7_pre20161022.ebuild [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x64-macos"
+IUSE="perl tcl unicode"
+
+DEPEND=">=sys-libs/db-4.2.52_p5:=
+	>=sys-libs/ncurses-5.6-r2:=
+	perl? ( dev-lang/perl )
+	tcl? ( >=dev-lang/tcl-8.5:0= )"
+
+BDEPEND="${DEPEND}
```
Comment (a17r) [inline app-editors/nvi/nvi-1.81.7_pre20161022.ebuild]: don't inherit `DEPEND` in there.

**Comment by doughdemon** [comment]
Thanks. Commit fixed.

> Does that also adress https://bugs.gentoo.org/656864 ?

I don't know, I was never able to reproduce this bug.

File: app-editors/nvi/nvi-1.81.7_pre20161022.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+IUSE="perl tcl unicode"
+
+DEPEND=">=sys-libs/db-4.2.52_p5:=
+	>=sys-libs/ncurses-5.6-r2:=
+	perl? ( dev-lang/perl )
+	tcl? ( >=dev-lang/tcl-8.5:0= )"
+
+BDEPEND="virtual/pkgconfig"
+
+RDEPEND="${CDEPEND}
```
Comment (a17r) [inline app-editors/nvi/nvi-1.81.7_pre20161022.ebuild]: seen this just now - `CDEPEND` does not exist in this ebuild. Presumably you want to inherit `DEPEND` there?

File: app-editors/nvi/nvi-1.81.7_pre20161022.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+REQUIRED_USE="tcl? ( !unicode )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.81.7-strlen-macro-renaming.patch
+	"${FILESDIR}"/${PN}-1.81.7-db44.patch
+	"${FILESDIR}"/${PN}-1.81.6-db.patch
+	"${FILESDIR}"/${PN}-1.81.6-printf-types.patch
+	"${FILESDIR}"/${PN}-1.81.7-distrib.patch
+	)
```
Comment (a17r) [inline app-editors/nvi/nvi-1.81.7_pre20161022.ebuild]: remove indent on closing brace

**Comment by doughdemon** [comment]
The tarball is missing some files that are not generated  by autoconf.
The distrib script does some more nonsense. I've now disabled a bit more.

**Comment by mgorny** [comment]
To be honest, this looks like a serious treecleaning candidate. It appears to have a lot of bugs, some of which you are only cheaply working around, and seems to be entirely dead upstream.

CC @gentoo/treecleaner

**Comment by doughdemon** [comment]
I have been in contact with upstream in the past but haven't heard
anything since 2017, and I don't have time to take over the project.

I think app-editors/nvi could be replaced by
>=app-editors/vim-8.1.1155[minimal]. The minimum version is important
because otherwise the multibyte support (an important feature of nvi)
is broken.

**Comment by juippis** [comment]
This is now package.masked and being removed in few days.

**Pull request closed without merging**